skimage/morphology/convex_hull.py

```diff
@@ -18,6 +18,56 @@ def _offsets_diamond(ndim):
     return offsets
 
 
+def _check_coords_in_hull(gridcoords, hull_equations, tolerance):
+    r"""Checks all the co-ordinates for inclusiveness in the convex hull.
+
+    Parameters
+    ----------
+    gridcoords : (M, N) ndarray
+        Co-ordinates of ``N`` points in ``M`` dimensions.
+    hull_equations : (M, N) ndarray
+        Hyperplane equations of the facets of the convex hull.
+    tolerance : float
+        Tolerance when determining whether a point is inside the hull. Due
+        to numerical floating point errors, a tolerance of 0 can result in
+        some points erroneously being classified as being outside the hull.
+
+    Returns
+    -------
+    coords_in_hull : ndarray of bool
+        Binary 1D ndarray representing points in n-dimensional space
+        with value ``True`` set for points inside the convex hull.
+
+    Notes
+    -----
+    Check inclusiveness of co-ordinates in convex hull requires intermediate
+    calculations of dot products which are memory-intensive. Thus, the convex
+    hull equations are checked individually with all co-ordinates to keep
+    within the memory limit.
+
+    References
+    ----------
+    .. [1] https://github.com/scikit-image/scikit-image/issues/5019
+
+    """
+    ndim, n_coords = gridcoords.shape
+    n_hull_equations = hull_equations.shape[0]
+    coords_in_hull = np.ones(n_coords, dtype=np.bool_)
+
+    # Pre-allocate arrays to cache intermediate results for reducing overheads
+    dot_array = np.zeros(n_coords, dtype=np.float64)
+    test_ineq_temp = np.zeros(n_coords, dtype=np.float64)
+
+    # A point is in the hull if it satisfies all of the hull's inequalities
+    for idx in range(n_hull_equations):
+        # Tests a hull equation on all co-ordinates of volume
+        np.dot(hull_equations[idx, :ndim], gridcoords, out=dot_array)
+        np.add(dot_array, hull_equations[idx, ndim:], out=test_ineq_temp)
+        coords_in_hull *= test_ineq_temp < tolerance
```

Comment:
This could also use a pre-allocated array using `np.less(test_ineq_temp, tolerance, out=coords_single_ineq)`. But this is less readable so we should make sure there is a measurable improvement in performance before doing this.

Reply:
> Does this work for your original problem and how does it compare in performance to the test case you mentioned? (5min vs 2min30s)?

Yes, and it took on average ~14 minutes. Also, one time there was a MemoryError of allocating space for an array with dimensions (ndim, n_coords) while computing the dot product, but I wasn't able to reproduce the error.

Reply:
> But this is less readable so we should make sure there is a measurable improvement in performance before doing this.

It took 18 seconds less for my test case

Reply:
> it took on average ~14 minutes

Yikes, that's not so great! 😂 It's ok though, I'm happy to get this merged like this and work on optimising the performance in a subsequent PR.

I suspect it might be worth applying the `np.less()` approach I suggested here for a gain in performance. Right now, for every loop, you need to allocate a massive array for `test_ineq_temp < tolerance`, and then, because you are hitting your RAM limit, deallocate it.

You can try to verify this with [`line_profiler`](https://github.com/pyutils/line_profiler).

Again, performance fix can come before or after this PR, I don't think it's important compared to getting this working.

@scikit-image/core anyone else up for approving this?

Reply:
Wait, I'll add the `np.less()` approach.